Tools/scripts/uploader.py

```diff
@@ -786,18 +791,23 @@ def ports_to_try(args):
         patterns = default_ports
     else:
         patterns = args.port.split(",")
+        # treat the specified port as a hint and add the default ports afterwards as on windows
+        # the port can change once the board is in bootloader mode
+        if "cygwin" in _platform or is_WSL:
```

Comment:
Did this get tested on cygwin?

Reply:
No, and there is a special place in hell for the python package manager and in that special place is an even more special place for cygwin and in that special place is an even more special place for the cygwin package manager. I thought I was finally free of cygwin, but now I've had to install all these weird things just to try this out.
Fortunately I seem to have got adept at sorting out python installation issues on my mac - skills which turn out to be transferable. Who knew?
It does not seem to work on cygwin, so I will take this part out.

Reply:
Fixed